pkg/virt-handler/notify-server/server.go

```diff
@@ -154,7 +155,23 @@ func RunServer(virtShareDir string, stopChan chan struct{}, c chan watch.Event,
 	case <-done:
 		log.Log.Info("notify server done")
 	case <-stopChan:
-		grpcServer.Stop()
+		gracefulShutdownTimeout := 10
+
+		isStopped := make(chan struct{})
+		go func() {
+			grpcServer.GracefulStop()
+			close(isStopped)
+		}()
+
+		t := time.NewTimer(time.Second * time.Duration(gracefulShutdownTimeout))
+		select {
+		case <-t.C:
+			log.Log.Infof("GracefulStop timed out after %d seconds, using Stop", gracefulShutdownTimeout)
+			grpcServer.Stop()
+		case <-isStopped:
+			t.Stop()
```

Comment:
You stopped the timer only here, so if there is indeed a timeout, it will leak until the gc kicks.
The timer should always be stopped.

You could have used `time.After()` here, but that depends as well on the gc and I would prefer to see a more deterministic approach here.

Reply:
We start a timer, and run GracefulStop, if the stop finished, it stop the timer, if the timer finished by itself its finished, what is please a 3rd option ?

Reply:
If the timeout has been reached, it will send on its channel and you will forcefully stop the grpc, but that does not stop the timer.

Reply:
a timeout doesn't of a timer doesn't mean it stopped ? i will take a look
(unless its a re triggering timer, where i understand)

Reply:
```
		t := time.NewTimer(time.Second * time.Duration(gracefulShutdownTimeout))
		select {
		case <-t.C:
			log.Log.Infof("GracefulStop timed out after %d seconds, using Stop", gracefulShutdownTimeout)
			grpcServer.Stop()
		case <-isStopped:
			t.Stop()
		}
```

Taken from grpc server.go
```
				timer := time.NewTimer(tempDelay)
				select {
				case <-timer.C:
				case <-s.quit.Done():
					timer.Stop()
					return nil
				}
```

they dont stop a stopped timer

Reply:
As I said earlier, this leaves the timer to the GC which is not a good practice.

Reply:
all the code does this, i will search for example or doc that say otherwise, need to even make sure it support Stop after TO

Reply:
Please give a proper explanation why you think it is not needed. The fact that in other places it is used like that is not an explanation.

Reply:
gc is something that we can relay on, in some aspects, this is one of them, 
but i will try to bring more info.
I think this is one of the examples, where PR shouldn't focus on such nits, which are well observed and widely used by even trusted 3rd parties, but just my opinion at least, and even if discussed, should not be blocking imo.

Reply:
> gc is something that we can relay on, in some aspects, this is one of them,
> but i will try to bring more info.
> I think this is one of the examples, where PR shouldn't focus on such nits, which are well observed and widely used by even trusted 3rd parties, but just my opinion at least, and even if discussed, should not be blocking imo.

I do not think it is a nit, and I expect a technical explanation. A reviewer can declare his comment a nit, not the other way around.